PWGMM/UE/Tasks/lambdak0sflattenicity.cxx

```diff
@@ -0,0 +1,826 @@
+      rFlattenicity.fill(HIST("hFT0Campl"), sumAmpFT0C);
+      rFlattenicity.fill(HIST("hFV0amplRing1to4"), sumAmpFV01to4Ch);
+      rFlattenicity.fill(HIST("hEv"), 4);
+      estimator[0] = multGlob;
+      estimator[1] = sumAmpFV0;
+      estimator[2] = 1.0 - flattenicityfv0;
+      estimator[3] = combined_estimator5;
+      float flatenicity_ft0 = (flatenicity_t0a + flatenicity_t0c) / 2.0;
+      estimator[4] = 1.0 - flatenicity_ft0;
+      estimator[5] = combined_estimator6;
+      float flatenicity_ft0v0 = 0.5 * flattenicityfv0 + 0.5 * flatenicity_t0c;
+      estimator[6] = 1.0 - flatenicity_ft0v0;
+      estimator[7] = ptT;
+      static_for<0, 7>([&](auto i) {
+        constexpr int index = i.value;
+        rFlattenicity.fill(HIST(nhEst[index]), estimator[index], estimator[0]); });
+      // plot pt vs estimators
+      for (auto& track : tracks) {
+        if (!track.isGlobalTrack()) {
+          continue;
+        }
+        float pt = track.pt();
+        static_for<0, 7>([&](auto i) {
+          constexpr int index = i.value;
+          rFlattenicity.fill(HIST(nhPtEst[index]), estimator[index], pt); });
+      }
+
+      for (int iCh = 0; iCh < 48; ++iCh) {
+        rFlattenicity.fill(HIST("hAmpV0VsCh"), iCh, ampchannel[iCh]);
+        rFlattenicity.fill(HIST("hAmpV0VsChBeforeCalibration"), iCh,
+                           ampchannelBefore[iCh]);
+      }
+      rFlattenicity.fill(HIST("fMultFv0"), sumAmpFV0);
+      rFlattenicity.fill(HIST("hFlatFT0CvsFlatFT0A"), flatenicity_t0c, flatenicity_t0a);
+    }
+
+    // ====================== Flattenicity estimation ends =====================
+
+    for (const auto& v0 : V0s) {
+      const auto& posDaughterTrack = v0.posTrack_as<TrackCandidates>();
+      const auto& negDaughterTrack = v0.negTrack_as<TrackCandidates>();
+
+      if (TMath::Abs(posDaughterTrack.eta()) > 0.8 || TMath::Abs(negDaughterTrack.eta()) > 0.8)
+        continue;
+      float massK0s = v0.mK0Short();
+      float massLambda = v0.mLambda();
+      float massAntiLambda = v0.mAntiLambda();
+
+      // float pdgmassK0s = 0.497614;
+      // float pdgmassLambda = 1.115683;
+
+      rKzeroShort.fill(HIST("hMassK0s"), massK0s);
+      rLambda.fill(HIST("hMassLambda"), massLambda);
+      rAntiLambda.fill(HIST("hMassAntiLambda"), massAntiLambda);
+
+      float decayvtxX = v0.x();
+      float decayvtxY = v0.y();
+      float decayvtxZ = v0.z();
+
+      float decaylength = TMath::Sqrt(TMath::Power(decayvtxX - vtxX, 2) + TMath::Power(decayvtxY - vtxY, 2) + TMath::Power(decayvtxZ - vtxZ, 2));
+      float v0p = TMath::Sqrt(v0.pt() * v0.pt() + v0.pz() * v0.pz());
+
+      float ctauK0s = decaylength * massK0s / v0p;
+      float ctauLambda = decaylength * massLambda / v0p;
+      float ctauAntiLambda = decaylength * massAntiLambda / v0p;
+
+      // Cut on dynamic columns for K0s
+
+      if (v0.v0cosPA() >= v0setting_cospaK0s) {                                       // cut on Cos of pointing angle
```

Comment:
Is there any reason why these conditions are separate and not a part of a single if?

Reply:
I have made it the part of single if.